src/borrow_tracker/stacked/mod.rs

```diff
@@ -929,14 +754,16 @@ trait EvalContextPrivExt<'mir: 'ecx, 'tcx: 'mir, 'ecx>: crate::MiriInterpCxExt<'
         // Note that this asserts that the allocation is mutable -- but since we are creating a
         // mutable pointer, that seems reasonable.
         let (alloc_extra, machine) = this.get_alloc_extra_mut(alloc_id)?;
-        let stacked_borrows = alloc_extra
-            .stacked_borrows
+        let mut stacked_borrows = alloc_extra
+            .borrow_tracker
             .as_mut()
             .expect("we should have Stacked Borrows data")
-            .get_mut();
+            .assume_sb()
+            .borrow_mut();
         let item = Item::new(new_tag, perm, protect.is_some());
         let range = alloc_range(base_offset, size);
-        let global = machine.stacked_borrows.as_ref().unwrap().borrow();
+        let global = machine.borrow_tracker_data.as_ref().unwrap().borrow();
+        // FIXME: can't share this with the current_span inside log_creation
```

Comment:
You are reverting some recent changes in master here. Please be careful when rebasing!

Reply:
`get_mut` doesn't work with `assert_sb` though. Or is that not the issue ?

Reply:
Ah, then you need `assert_sb_mut`.